### PR TITLE
[Backport 2.x] Adding concurrent search versions of query count and time metrics (#9622)

### PR DESCRIPTION
**Pull request by jed326**
### Description
2.x Backport PR for concurrent segment search related metrics
- #9622 
- https://github.com/opensearch-project/OpenSearch/pull/9670

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/index/search/stats/SearchStats.java
Comment (reta) [inline server/src/main/java/org/opensearch/index/search/stats/SearchStats.java:162]: 👍

File: server/src/main/java/org/opensearch/index/search/stats/SearchStats.java
Comment (reta) [inline server/src/main/java/org/opensearch/index/search/stats/SearchStats.java:342]: 👍

File: server/src/main/java/org/opensearch/index/search/stats/SearchStats.java
Comment (reta) [reply server/src/main/java/org/opensearch/index/search/stats/SearchStats.java:162]: @jed326 could you please create a pull request to main with these version changes only (in & out), we'll merge them in lockstep, thank you.

**Review by reta — APPROVED** [review]

File: server/src/main/java/org/opensearch/index/search/stats/SearchStats.java
Comment (jed326) [reply server/src/main/java/org/opensearch/index/search/stats/SearchStats.java:162]: @reta I'm about to publish a PR for https://github.com/opensearch-project/OpenSearch/issues/9644, would it be okay if I got that change merged into main first and then backport both at the same time? That way we don't have to deal with bwc backporting multiple times

Comment (reta) [reply server/src/main/java/org/opensearch/index/search/stats/SearchStats.java:162]: > @reta I'm about to publish a PR for #9644, would it be okay if I got that change merged into main first and then backport both at the same time? That way we don't have to deal with bwc backporting multiple times

Sure, but we cannot merge this one than, cuz if we do, all BWC tests on `main` would start to fail.

Comment (jed326) [reply server/src/main/java/org/opensearch/index/search/stats/SearchStats.java:162]: Yep makes sense, let me convert this PR back to draft and then i'll re-use the same branch when backporting both commits.

**Comment by jed326** [comment]
Will use this PR to backport #9670 as well.

**Comment by jed326** [comment]
Hey @reta now that https://github.com/opensearch-project/OpenSearch/pull/9670 is merged I updated this PR to do the backport for both concurrent segment search metrics PRs.
As requested, I also opened https://github.com/opensearch-project/OpenSearch/pull/9722 as well to apply these bwc changes on main at the same time. Could you please help review & merge both? Thanks!

**Pull request merged**
